plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+	}
+
+	/**
+	 * Process data for the PublicationFile
+	 *
+	 * @param object $data
+	 * @param int $submissionId
+	 * @param string $filePath
+	 * @param string $extension
+	 * @param int $publicationFormatId
+	 * @param int $pressId
+	 * @param int $genreId
+	 */
+	private function processPublicationFile($data, $submissionId, $filePath, $extension, $publicationFormatId, $pressId, $genreId) {
+		$submissionDir = sprintf($this->_format, $pressId, $submissionId);
+		$fileId = $this->_fileService->add($filePath, $submissionDir . '/' . uniqid() . '.' . $extension);
+		$mimeType = PKPString::mime_content_type($filePath);
+
+		/** @var SubmissionFileDAO $submissionFileDao */
+		$submissionFileDao = $this->getCachedDao('SubmissionFileDAO');
+
+		/** @var SubmissionFile $submissionFile */
+		$submissionFile = $submissionFileDao->newDataObject();
+		$submissionFile->setData('submissionId', $submissionId);
+		$submissionFile->setData('uploaderUserId', $this->_userId);
+		$submissionFile->setSubmissionLocale($data->locale);
+		$submissionFile->setGenreId($genreId);
+		$submissionFile->setFileStage(SUBMISSION_FILE_PROOF);
+		$submissionFile->setAssocType(ASSOC_TYPE_REPRESENTATION);
+		$submissionFile->setData('assocId', $publicationFormatId);
+		$submissionFile->setData('createdAt', Core::getCurrentDate());
+		$submissionFile->setDateModified(Core::getCurrentDate());
+		$submissionFile->setData('mimetype', $mimeType);
+		$submissionFile->setData('fileId', $fileId);
+		$submissionFile->setData('name', pathinfo($filePath, PATHINFO_FILENAME));
+
+		// Assume open access, no price.
+		$submissionFile->setDirectSalesPrice(0);
+		$submissionFile->setSalesType('openAccess');
+
+		$submissionFileDao->insertObject($submissionFile);
+	}
+
+	/**
+	 * Process data for Keywords
+	 *
+	 * @param object $data
+	 * @param int $publicationId
+	 */
+	private function processKeywords($data, $publicationId) {
+		$keywordsList = [$data->locale => explode(';', $data->keywords)];
+
+		if (count($keywordsList[$data->locale]) > 0) {
+			/** @var SubmissionKeywordDAO $submissionKeywordDao */
+			$submissionKeywordDao = $this->getCachedDao('SubmissionKeywordDAO');
+			$submissionKeywordDao->insertKeywords($keywordsList, $publicationId);
+		}
 	}
-}
 
+	/**
+	 * Process data for Subjects
+	 *
+	 * @param object $data
+	 * @param int $publicationId
+	 */
+	private function processSubjects($data, $publicationId) {
+		$subjectsList = [$data->locale => explode(';', $data->subjects)];
+
+		if (count($subjectsList[$data->locale]) > 0) {
+			/** @var SubmissionSubjectDAO $submissionSubjectDao */
+			$submissionSubjectDao = $this->getCachedDao('SubmissionSubjectDAO');
+			$submissionSubjectDao->insertSubjects($subjectsList, $publicationId);
+		}
+	}
+
+	/**
+	 * Process data for the Book Cover Image
+	 *
+	 * @param object $data
+	 * @param string $srcFilePath
+	 * @param Publication $publication
+	 */
+	private function processBookCoverImage($data, $srcFilePath, $publication) {
```

Comment:
It's ok to type the inputs and returns of the methods, just be careful with the types that can be `null`. But maybe it's better to leave this for the `stable_3_4_0`.